assignment-1/Functions.java

```diff
@@ -0,0 +1,271 @@
+        }
+    }
+
+    //Fourth Task
+    public static void searchPlayer(String playerName) {
+        try {
+            boolean flag = true;
+            for(Players p : allPlayers) {
+                if(p.getName().contains(playerName))
+                {
+                    flag = true;
+                    System.out.println("Name: " + p.getName());
+                    System.out.println("Team: " + p.getTeam());
+                    System.out.println("Role: " + p.getRole());
+                    System.out.println();
+                }
+                else {
+                    flag = false;
+                }
+            }
+            if(!flag) {
+                System.out.println("No Player found with name " + playerName);
+            }
+        } catch (Exception e) {
+            System.out.println(e.getMessage());
+        }
+    }
+
+//    Fifth Task
+
+    public static void runsAndWicketsScorer(String teamName) {
+
+        HashMap<String, Integer> runScorer = new HashMap<>();
+        HashMap<String, Integer> wicketTaker = new HashMap<>();
+        Map<String, Integer> sorted = new HashMap<>();
+        try {
+            for (Players player : allPlayers) {
+                if (player.getTeam().equals(teamName)) {
+                    runScorer.put(player.getName(), player.getRuns());
+                }
+                if (player.getTeam().equals(teamName)) {
+                    wicketTaker.put(player.getName(), player.getWickets());
+                }
+            }
+            Map<String, Integer> map = new HashMap<>();
+
+            sorted = runScorer.entrySet().stream().sorted(Collections.reverseOrder(Map.Entry.comparingByValue())).collect(toMap(Map.Entry::getKey, Map.Entry::getValue, (e1, e2) -> e2, LinkedHashMap::new));
+            Map.Entry<String, Integer> entry = sorted.entrySet().iterator().next();
+            System.out.println("Highest Run Scorer");
+            System.out.println(entry.getKey() + " " + entry.getValue());
+
+            sorted = wicketTaker.entrySet().stream().sorted(Collections.reverseOrder(Map.Entry.comparingByValue())).collect(toMap(Map.Entry::getKey, Map.Entry::getValue, (e1, e2) -> e2, LinkedHashMap::new));
+            Map.Entry<String, Integer> entry1 = sorted.entrySet().iterator().next();
+            System.out.println("\nHighest Wicket Taker");
+            System.out.println(entry1.getKey() + " " + entry1.getValue());
+        } catch (Exception e) {
+            e.getMessage();
+        }
+    }
+
+    //Sixth Task
+    public static void top3BatBowlAllRounder() {
+        System.out.println("\nTop 3 Batsman of the season");
+        try {
+            List<Players> topBatters = allPlayers.stream()
+                    .sorted(Comparator.comparingInt(Players::getRuns).reversed()).limit(3)
+                    .collect(Collectors.toList());
+            for (Players lp : topBatters) {
+                System.out.println(lp.getName());
```

Comment:
if you had overridden the toString method, you could have directly printed the player object here